### ISSUE TITLE
Map config item with the map values gets the wrong keys

### ISSUE THREAD

**Issue by sberyozkin** [quarkusio/quarkus#5982]
**Describe the bug**

With this config item:
```
/**
     * Configuration of the individual parsers.
     * For example:
     *
     * <pre>
     * quarkus.tika.parsers = pdf
     * quarkus.tika.parser-options.pdf.sort-by-position = true
     */
    @ConfigItem
    public Map<String, Map<String, String>> parserOptions;
```

the first segment following the name of the config item (`pdf`) is lost. 

Before the major config update I was seeing the map initialized as follows, given the above config fragment:

`pdf` (key):  `sort-by-position` = `true` (single entry map value, `sort-by-position` - key, `true` - value)

Now I'm seeing:

`sort-by-position` (key):  `sort-by-position` = `true` (single entry map value, `sort-by-position` - key, `true` - value)

so the `pdf` bit is lost.

(I think a similar issue may affect [this configuration](https://github.com/quarkusio/quarkus/blob/master/extensions/keycloak-authorization/runtime/src/main/java/io/quarkus/keycloak/pep/runtime/KeycloakPolicyEnforcerConfig.java#L164))

CC @dmlloyd

**Comment by gsmet** [comment]
@dmlloyd Dunno how hard it will be to get this fixed but we would probably need a fix for 1.1.0.

**Comment by dmlloyd** [comment]
I wrote a test for this case... and the test passes.  The keys are correctly mapped.  I'll keep trying to reproduce though.

**Comment by sberyozkin** [comment]
@dmlloyd You can reproduce it real fast in `integration-tests/tika` with `mvn test -Dtest=TikaPdfInvoiceTest` in #4360.

**Comment by dmlloyd** [comment]
I have a fix, but #5989 has to be merged first due to conflicts.

**Comment by sberyozkin** [comment]
@dmlloyd thanks, I will follow up with the #4360 rebase after the fix is in
